### ISSUE TITLE
Additon of channel assignment graphics in wizard

### ISSUE THREAD

**Issue by Dsrenger** [FrSkyRC/ETHOS-Feedback-Community#4753]
As discussed in https://github.com/FrSkyRC/ETHOS-Feedback-Community/issues/4740 here is now a new discussion around the channel assignment

**Reason for the issue:**
The assignment of the servos to the correct channels has a major impact to the programming later on like in differential setups.
There is some confusion around how Ethos expects the channel assignment. Especially as there is some FrSky documentation (SR receivers) that are contradicting.
Right now, neither Ethos nor the manual is showing the expected channel assignment.

**Suggestion for solution**
There could be a possibility to show the expected channel assignment in Ethos in the model creation wizard

**Impact to customers**
As most of the fixed wing pilots are creating differential setups, the channel assignment is important for most of them. The implementation could reduce a lot of frustration for them.

Disclaimer:
As I dont know the correct channel assigment either, the mock ups I show now are just examples. I have no idea if I choose the correct assignment. An Ethos developer would need to correct if needed

Currently there is a lot of space in the Channel assignment screen:

![image](https://github.com/user-attachments/assets/ef3d570f-51e8-47f2-a081-25a3f2251223)

It could be possible to reuse the graphics from one page before and place a text at the correct positions to indicate the correct assignment

It could look like this (2 random examples, but I think it makes the concept clear). As said I have no idea if these assginments are correct or not ;)
![image](https://github.com/user-attachments/assets/ebdf4b5c-54a8-462a-b906-a8c7f6fdebbc)
![image](https://github.com/user-attachments/assets/dd54306e-4b5e-4755-83df-4b68db6636c5)

The challenge will be to make it fit for multiple languages, screen sizes and number of flaps/elevators ect.
As I think, that there are no language specific graphics at the moment, I suggested to have language neutral graphics and place texts around it. These texts can be language specific through the i18n file/translations.
But I could imagine as well a color coding
![image](https://github.com/user-attachments/assets/847c4718-50b1-4e3d-9745-78c0947e3ee2)

To be honest, I have no idea how to excactly execute a concept like this, as I am no software programmer. 
I leave the best implementation up to you, as you are the experts.

**Comment by hrenz** [comment]
Example with a SR12
![image](https://github.com/user-attachments/assets/f39d9cd9-e298-4aad-bf49-13f742076fd7)

**Comment by hrenz** [comment]
Example with a SR8
![image](https://github.com/user-attachments/assets/3f4cdde7-6c84-4443-84fa-d40ad8621f0b)

**Comment by hrenz** [comment]
Example with SR8 2QR, 2FL, 1SR 1HR, 1 Motor,  1 Kanal frei für Fahrwerk
![image](https://github.com/user-attachments/assets/2150e717-12b2-4f9d-8b55-4dfa0f7f39c6)

**Comment by Dsrenger** [comment]
It makes no sense to write the channel into/nearby the graphic.

The assigned channel is showed in the table below.

The graphic should just show which column should correspond to which flap.

**Comment by hrenz** [comment]
It makes sense, then you have a universal model, whether with or without gyro
It is always the same channel assignment for you, which simplifies many things
You can then simply copy the model and deselect what you don't need. 

![image](https://github.com/user-attachments/assets/4f7ca007-70c1-4e0d-970f-6429ed9101e6)

![image](https://github.com/user-attachments/assets/83e91cb7-229d-42a3-a9f7-87708f243c8a)

**Comment by hrenz** [comment]
Thinking ahead:
But to make it uniform for youself, I always do it as if a gyro receiver were installed.
Then you don't have to change anything and it's the same for all models, whether with or without a gyro.
and can copy models, in the sense of a universal model where only what you need is locked/released.
Even when retrofitting/converting to gyros, nothing needs to be changed, which simplifies many things.

**Comment by hrenz** [comment]
That would probably be the simplest version with SR8, I think that's enough for 80% of hobby pilots
Whether with or without gyros, the channel assignment is always the same and you can simply copy the models
regardless of whether you have flaps or not. I think this is ideal for an ETHOS newcomer
And one channel (Ch8) is still free for extending and retracting the landing gear

![image](https://github.com/user-attachments/assets/57481073-6b59-49b6-be4f-0ce17fb3c97e)

**Comment by strgaltdel** [comment]
Just a few additional thoughts on Dennis' description:

The goal is to enable the user to recognize the assignment of a **SPECIFIC** function (e.g., OUTER, LEFT aileron) in the wizard in a simple way.

The previous representation only conveyed the group function (aileron) and numbered the flaps sequentially. A specific assignment (left or right / inner outer flap) for later processing in mixers was not conveyed.

In my experience, as Dennis also pointed out, users can be easily "guided" by the association of colors.
However, I cannot rule out that my following idea might introduce too much color into the mix—it's simply another idea:

The "color assignment" follows consistent rules:

- Functions within a group (aileron, camber-flaps, elevator, etc.) are assigned to a color family (red, blue, green) to better distinguish groups (basic functionality).
- Within a group (e.g., ailerons), paired surfaces fade from solid tones (outer or main control surfaces) to lighter shades.
- Within a pair (e.g., outer ailerons), the right control surface has the saturated tone, while the left control surface has the lighter shade.

As a result, an example color scheme could look like this:

![image](https://github.com/user-attachments/assets/f4d88b7a-bfbc-4932-b569-693985ff29d1)




- Additionally, the function name will also include "r" or "l" for right or left control surface.
- Labels will also be displayed in the graphic.
- When editing a channel assignment, the label can appear in red


In summary, it could look like this (Ail4l was just edited):

![image](https://github.com/user-attachments/assets/83d349b0-50cf-4e84-b204-b6685f837ae3)

My choice of colors was arbitrary; a UI expert could probably do it better (-;

**Comment by hrenz** [comment]
I think the error occurs in ETHOS here,
in the assignment of the aileron mixers to the channels K1 and K5

Mixer Output 1: is the positive output,  goes to the left aileron K1, but should go to right aileron K5
Mixer Output 2: is the negative output, goes to the right aileron K5, but should go to the left aileron K1.

Example: hold Aileron stick full right 
Weighting100%, Differentiation +30%
K5 must go up +100%
K1 must go down -70%
Then everything would be correct

But exactly the opposite happens

This means that the sequence in the aileron mixer would have to be reversed
Instead of 
Mixer Output 1 to K1 
Mixer Output 2 to K5
It would have to be 
Mixer Output 1 to K5
Mixer Output 2 to K1

Then everything would be correct again

The channel sequence from left to right K1 left, K5 right
The wizzards can stay from left to right
The gyro assignment from left to right
The layout would still be correct 
The instructions and manuals would continue to be correct 
The end user would not notice anything

Please think about it
All gyro instructions go from left K1 to right K5 
I think this error has been hidden in ETHOS for a long time

![image](https://github.com/user-attachments/assets/b54fd1d4-2d3f-4983-afdd-03dffabd6aa7)

![image](https://github.com/user-attachments/assets/b599d513-229b-449b-8092-faaf75d148c1)

![image](https://github.com/user-attachments/assets/3d65e379-763a-490d-8e34-c39c307b030e)

**Comment by hrenz** [comment]
think this is the correkt way
K1 Ail1 Left to  K5 Ail2 Right then alle calculations are correct

![image](https://github.com/user-attachments/assets/ec15e1d5-c45a-43fc-8054-66b868f4ff89)

**Comment by bsongis-frsky** [comment]
I understand nothing in the graphics above. I am not sure it will help the user!

**Comment by Dsrenger** [comment]
Think so too.

I think strgaltdel made the best proposal so far.

@bsongis-frsky 
Just one heads-up, as there were a bigger discussion in the German forum around this.
We assume that Ethos expects the right aileron as aileron1. That fits the "rule" that positive values let the flap go up. Then the differential is as well positive so that the aileron that goes down will go down less 

The confusion came up, as this is contrary to the manual of the gyro receivers. They are expecting aileron1 as the left airleron. This leads to a negative value of the differential.

Not a big deal by itself, but be careful how to show the channel assignment in Ethos. There's a risk that it will be contrary to the gyro Rx manuals.
Guess there is some kind of disconnect. 

This just as heads up if this really gets implemented.

That's the reason why I didn't care for the channel assignment in my graphics. As I don't know exactly how you really expect it in Ethos. Right aileron to ail1 was just a educated guess.

**Comment by bsongis-frsky** [comment]
For now we will keep your very first proposal. With the gyro receivers we will adapt the names, they will be written in the drawing too. We cannot change that or we would have unhappy users!

**Comment by Dsrenger** [comment]
Ok, so you had it on your radar. Just wanted to give a heads up if that would not be the case. 
As you said, there was a potential for a lot of unhappy Users. Changing something there is not a good idea. I agree. Wanted to make sure you are aware. 

I am pretty sure you will implement it fine. Will have a look at it, once it's in the nightlies to check what's the implications of the name changes are. I am bad at imagine it without testing ;)

**Comment by Dsrenger** [comment]
Just one last word. 

Main focus HAS to be, that be existing model doesnt need a change. But I think that is, what you meant with "unhappy users"

If there will be a substantial change for the users, it might be better to leave everything as is.

**Comment by bsongis** [comment]
Thanks. And yes I will be very careful with that feature, it is extremely easy to break things.

**Comment by hrenz** [comment]
Gyro K1 Left   K5 Right
![image](https://github.com/user-attachments/assets/b78df292-5416-41c8-a36f-f063a7fd5957)
